mentat/session.py

```diff
@@ -57,8 +57,7 @@ def __init__(
 
         # Since we can't set the session_context until after all of the singletons are created,
         # any singletons used in the constructor of another singleton must be passed in
-        # TODO: An error is thrown in this function; once git root is removed, the error will be removed
-        git_root = get_shared_git_root_for_paths([Path(path) for path in paths])
```

Comment:
In config.py on line 150 we use this function to get the git root so that we can check for the project config; maybe you were already planning to do this in another PR, but the get_shared_git_root function will currently crash. I think we could change it to the get_git_root_for_path function and we could first check for a project config in the git root and then in the cwd if there is not git root.

Reply:
> I think we could change it to the get_git_root_for_path function and we could first check for a project config in the git root and then in the cwd if there is not git root.

Agree on this being the behavior for determining the config to use. I'll add this in another PR!